src/beamlime/config/models.py

```diff
@@ -179,3 +179,110 @@ def from_string(cls, key_str: str) -> ConfigKey:
         if service_name == '*':
             service_name = None
         return cls(source_name=source_name, service_name=service_name, key=key)
+
+
+T = TypeVar('T')
+
+
+class ParameterType(str, Enum):
+    """
+    Enum for parameter types.
+
+    This enum is used to define the type of a parameter.
+    """
+
+    INT = 'int'
+    FLOAT = 'float'
+    STRING = 'string'
+    BOOL = 'bool'
+    OPTIONS = 'options'
+
+
+class Parameter(BaseModel, Generic[T]):
+    """
+    Model for workflow parameter.
+
+    This model is used to define a parameter for a specific workflow.
+    """
+
+    name: str = Field(description="Name of the parameter.")
+    unit: str | None = Field(default=None, description="Unit of the parameter.")
+    description: str = Field(description="Description of the parameter.")
+    param_type: ParameterType = Field(description="Type of the parameter.")
+    default: T | None = Field(
+        default=None, description="Default value of the parameter."
+    )
+    options: list[Any] | None = Field(
+        default=None,
+        description="List of options, if it is a parameter with options.",
+    )
+
+    @model_validator(mode='after')
+    def validate_options(self) -> Parameter:
+        """Validate that options are provided for OPTIONS parameter type."""
+        if self.param_type == ParameterType.OPTIONS:
+            if self.options is None or not self.options:
+                raise ValueError(
+                    "Options must be provided for parameter type 'OPTIONS'"
+                )
+            if self.default not in set(self.options):
```

Comment:
```suggestion
            if self.default not in self.options:
```
?

Reply:
I think I got linter complaints without the `set`.